### ISSUE TITLE
Group: window list does not bring window into focus on click

### ISSUE THREAD

**Issue by leousa** [Simmesimme/Gnome-Pie#98]
In Ubuntu 14.04 when creating a pie for Group: window lists (like alt+tab), clicking on a slice doesn't bring the corresponding window into focus. The window is highlighted in the Unity launcher, but doesn't come forefront into focus. Tested in 2 different computers running 14.04.

Steps to replicate:

1- Create custom pie
2- Select Group:window lists
3- Open several windows on computer, for example, nautilus, firefox, writer, etc.
4- Call Gnome-pie custom launcher.
5- Click on any window that is not into focus.
6- On click, the window is highlighted in the Unity launcher menu, but it does not come into focus

Expected behaviour:
6- On click the window comes forefront into focus.

**Comment by Simmesimme** [comment]
Thanks for your report! Can you please exit Gnome-Pie and launch it from terminal by typing

`gnome-pie`

Is anything printed to the terminal when activating a slice of your window-list-pie?

**Comment by ronaldoeyeem** [comment]
I came to the project to report on the same issue, I had it both on Ubuntu 14.04 and 14.10 (both at work n home). 
Home PC is intel i5 with real time kernel
Work PC is intel i7 with generic kernel
No further customization on the systems. Ubuntu with unity and removed unnecessary stuff like shopping lens.


the logs are:

```
$ gnome-pie
[MESSAGE] Welcome to Gnome-Pie 0.6.1!
[MESSAGE] Loading Pies from "/home/budius/.config/gnome-pie/pies.conf".
[MESSAGE] Started happily...
[MESSAGE] Devices changed...
[WARNING] Failed to fetch xid: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface 'org.ayatana.bamf.window' on object at path /org/ayatana/bamf/window/52428885
[WARNING] Failed to fetch type: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface 'org.ayatana.bamf.window' on object at path /org/ayatana/bamf/window/52428885
[WARNING] Failed to fetch monitor: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface 'org.ayatana.bamf.window' on object at path /org/ayatana/bamf/window/52428885
[WARNING] Failed to fetch maximized state: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface 'org.ayatana.bamf.window' on object at path /org/ayatana/bamf/window/52428885
[WARNING] Failed to fetch xid: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface 'org.ayatana.bamf.window' on object at path /org/ayatana/bamf/window/52428885
[WARNING] Unable to fetch children: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface 'org.ayatana.bamf.view' on object at path /org/ayatana/bamf/application/1339296950

[WARNING] Failed to fetch path: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface 'org.ayatana.bamf.application' on object at path /org/ayatana/bamf/application/1339296950

(gnome-pie:6014): Wnck-WARNING **: Received a timestamp of 0; window activation may not function properly.

```

I hope it helps, I'm loving this project, totally replaced my unity launcher

**Comment by leousa** [comment]
Here is my log when launched from terminal and trying to bring window into focus using the gnome-pie launcher
leo@Lemur:~$ gnome-pie
[MESSAGE] Welcome to Gnome-Pie 0.6.1!
[MESSAGE] Loading Pies from "/home/leo/.config/gnome-pie/pies.conf".
[WARNING] Failed to find file ".gtk-bookmarks"!
[WARNING] Icon "gnome-session-halt" not found! Using default icon...
[WARNING] No default icon found! Will be ugly...
[MESSAGE] Error loading image file: Failed to open file '': No such file or directory
[WARNING] Icon "stock_unknown" not found! Using default icon...
[WARNING] No default icon found! Will be ugly...
[MESSAGE] Error loading image file: Failed to open file '': No such file or directory
[MESSAGE] Started happily...

(gnome-pie:12517): Wnck-WARNING **: Received a timestamp of 0; window activation may not function properly.
(gnome-pie:12517): Wnck-WARNING **: Received a timestamp of 0; window activation may not function properly.
(gnome-pie:12517): Wnck-WARNING **: Received a timestamp of 0; window activation may not function properly.

**Comment by Simmesimme** [comment]
Ok... this Wnck-WARNING seems interesting, I will have a look at this!

**Comment by Simmesimme** [comment]
Hey, there was a simple fix I guess (9af18c88c382fe10bd0e8818332759da6f3d5495). Is it possible for one of you to compile the develop branch and confirm that it works as supposed?

**Comment by gabdub** [comment]
Solve the issue for me in a VM running 14.04 32bits.
I'll tested in 14.04 64-bits when I'm back at home.

**Comment by ronaldoeyeem** [comment]
I'm installing dependencies compiling at this moment. I'll report back later today.

Do we get an ETA on this patch or I better use the dev branch for now?

**Comment by ronaldoeyeem** [comment]
`OK` tested on my work PC.
stock Ubuntu with Unity 14.10 (x64) on i7.
All working good.

Still doesn't change workspace if the running application is not in current workspace (but I guess that would be a different ticket).

**Comment by leousa** [comment]
An opt-in option to change workspaces if windows in different workspace would be amazing.

**Comment by Simmesimme** [comment]
This fix will be part of 0.6.2 which I'll try to release at the end of next week!
